WebRoot/js/zimbraAdmin/servers/controller/ZaServerController.js

```diff
@@ -165,8 +165,46 @@ function () {
 ZaServerController.prototype._saveChanges =
 function () {
 	var obj = this._view.getObject();
-    if (this._currentObject[ZaModel.currentTab]!= obj[ZaModel.currentTab])
-             this._currentObject[ZaModel.currentTab] = obj[ZaModel.currentTab];
+    if (this._currentObject[ZaModel.currentTab]!= obj[ZaModel.currentTab]) {
+		this._currentObject[ZaModel.currentTab] = obj[ZaModel.currentTab];
+	}
+		 
+	//Check validation of reverse proxy upstream imap server: host name or IP address format
+	var invalidValues = [];
+	var isValidValue = true;
+	
+	if (ZaItem.hasWritePermission(ZaServer.A_zimbraReverseProxyUpstreamImapServers,obj)) {
```

Comment:
isn't this same as ZaServerController.prototype.validateReverseProxyUpstreamImapServers ?

Reply:
this doesn't seem fixed yet